teachertool/src/components/ProjectWorkspace.tsx

```diff
@@ -0,0 +1,27 @@
+import * as React from "react";
+// eslint-disable-next-line import/no-internal-modules
+import css from "./styling/ProjectWorkspace.module.scss";
+
+import { Toolbar } from "./Toolbar";
+import { AddressBar } from "./AddressBar";
+import { MakeCodeFrame } from "./MakecodeFrame";
+import { classes } from "../utils";
+
+interface IProps {}
+
+export const ProjectWorkspace: React.FC<IProps> = () => {
+    return (
+        <div className={classes(css, "panel", "h-full", "w-full", "flex", "flex-col")}>
+            <Toolbar toolbarClass={css["grow-1"]}>
```

Comment:
They actually can still validate a project even after switching to JS. It'll still validate the blocks, but it should all still work. It won't work well if a project was **made** in Javascript instead of blocks, but I think that's a different scenario.

Reply:
Yeah, I'm worried the toggle will give the impression that we support evaluating JavaScript-only projects, but this is an assumption. I imagine a scenario where a teacher sees the toggle, thinks "Oh, I can validate student projects in JavaScript, too", uploads a JavaScript-only project and then the evaluation doesn't work. But, if a teacher uploads a JavaScript project, that's something that we can know, right? And just error and tell the user "sorry, we cannot evaluate JavaScript-made projects at this time".

Reply:
I wouldn't worry about it too much for v1, but yes I think we can check the preferred editor and error if needed.

I think there is value in having the toggle regardless, as one teacher did mention they like to switch to JS to search for things in the code, rather than scrolling through blocks, so there's a known use-case there.